components/messenger/transports.rst

```diff
@@ -0,0 +1,225 @@
+in the code repository.
+
+Enqueue
+-------
+
+Probably one of the most famous PHP queue-broker libraries, Enqueue, has 10+ adapters
+with brokers like Kafka, Google Pub/Sub, AWS SQS and more. Check out the transport
+documentation in `Enqueue's official repository`_.
+
+Your own transport
+------------------
+
+If there is no available transport for your message broker, you can easily
+create your own.
+
+Your own sender
+~~~~~~~~~~~~~~~
+
+Using the ``SenderInterface``, you can easily create your own message sender.
+Let's say you already have an ``ImportantAction`` message going through the
+message bus and handled by a handler. Now, you also want to send this message as
+an email.
+
+First, create your sender::
+
+    namespace App\MessageSender;
+
+    use App\Message\ImportantAction;
+    use Symfony\Component\Messenger\Transport\SenderInterface;
+
+    class ImportantActionToEmailSender implements SenderInterface
+    {
+       private $toEmail;
+       private $mailer;
+
+       public function __construct(\Swift_Mailer $mailer, string $toEmail)
+       {
+           $this->mailer = $mailer;
+           $this->toEmail = $toEmail;
+       }
+
+       public function send($message)
+       {
+           if (!$message instanceof ImportantAction) {
+               throw new \InvalidArgumentException(sprintf('Producer only supports "%s" messages.', ImportantAction::class));
+           }
+
+           $this->mailer->send(
+               (new \Swift_Message('Important action made'))
+                   ->setTo($this->toEmail)
+                   ->setBody(
+                       '<h1>Important action</h1><p>Made by '.$message->getUsername().'</p>',
+                       'text/html'
+                   )
+           );
+       }
+    }
+
+Your own receiver
+~~~~~~~~~~~~~~~~~
+
+A receiver is responsible for receiving messages from a source and dispatching
+them to the application.
+
+Let's say you already processed some "orders" in your application using a
+``NewOrder`` message. Now you want to integrate with a 3rd party or a legacy
+application but you can't use an API and need to use a shared CSV file with new
+orders.
+
+You will read this CSV file and dispatch a ``NewOrder`` message. All you need to
+do is to write your custom CSV receiver and Symfony will do the rest.
+
+First, create your receiver::
+
+    namespace App\MessageReceiver;
+
+    use App\Message\NewOrder;
+    use Symfony\Component\Messenger\Transport\ReceiverInterface;
+    use Symfony\Component\Serializer\SerializerInterface;
+
+    class NewOrdersFromCsvFile implements ReceiverInterface
+    {
+       private $serializer;
+       private $filePath;
+
+       public function __construct(SerializerInteface $serializer, string $filePath)
+       {
+           $this->serializer = $serializer;
+           $this->filePath = $filePath;
+       }
+
+       public function receive(callable $handler) : void
```

Comment:
```suggestion
       public function receive(callable $handler)
```

or is the `: void` part of the interface?

Reply:
@OskarStark there is code changes in closed PR with `:void`
https://github.com/symfony/symfony/pull/27008/files#diff-28db5c7ca1e1eca1623ad72ddd29b268L38

Reply:
@OskarStark It's ok to add a return type hint even if it isn't in the interface. I'd even say it is recommended for static analysis.

Reply:
I know but O am not sure how we handle this in docs in general 🤔

Ping @symfony/team-symfony-docs

Reply:
Wait, isn’t receive gone from the interface now? I think it’s get()